community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/input/BadCollector.java

```diff
@@ -56,16 +68,44 @@ public BadCollector( OutputStream out, int tolerance, int collect )
     }
 
     @Override
-    public void collectBadRelationship( InputRelationship relationship, Object specificValue )
+    public void collectBadRelationship( final InputRelationship relationship, final Object specificValue )
     {
-        checkTolerance( BAD_RELATIONSHIPS,
-                new InputException( format( "%s refering to missing node %s", relationship, specificValue ) ) );
+        checkTolerance( BAD_RELATIONSHIPS, new Bad()
```

Comment:
New instance of `Bad` will be created for each call which is not a memory saver, as `Bad` trying to be. Looks like exceptions should be create only in case we already exceed tolerance. WDYT?